src/main/kotlin/io/libp2p/etc/types/Delegates.kt

```diff
@@ -55,20 +60,20 @@ class LazyMutable<T>(val initializer: () -> T, val rejectSetAfterGet: Boolean =
 }
 
 data class CappedValueDelegate<C : Comparable<C>>(
-    var value: C,
-    val lowerBound: C?,
-    val lowerBoundVal: C? = lowerBound,
-    val upperBound: C?,
-    val upperBoundVal: C? = upperBound
-) :
-    ReadWriteProperty<Any?, C> {
+    private var value: C,
+    private val lowerBound: () -> C,
+    private val lowerBoundVal: () -> C = lowerBound,
+    private val upperBound: () -> C,
+    private val upperBoundVal: () -> C = upperBound
+) : ReadWriteProperty<Any?, C> {
 
     override fun getValue(thisRef: Any?, property: KProperty<*>): C {
+        val v1 = if (value > upperBound.invoke()) upperBoundVal.invoke() else value
+        value = if (value < lowerBound.invoke()) lowerBoundVal.invoke() else v1
```

Comment:
Hm, modifying the `value` in getter looks slightly weird but it does the trick with dynamic bounds

Reply:
Yep, I saw this case and just could not invent anything better. 
> Which seems weird?

Just update the value in getter seems strange. 
Sorry if I maybe misused the word 'weird'. Not sure probably it sounds rather 'awful' than 'strange' :)